pkg/logs/message/message.go

```diff
@@ -5,25 +5,46 @@
 
 package message
 
+import "github.com/DataDog/datadog-agent/pkg/logs/config"
+
 // Message represents a log line sent to datadog, with its metadata
 type Message struct {
 	Content    []byte
 	Origin     *Origin
-	status     string
-	Timestamp  string
 	RawDataLen int
+	Timestamp  string
+
+	status string
+}
+
+// NewPartialMessage3 constructs message with content, status and origin.
+func NewPartialMessage3(content []byte, status string, origin *Origin) *Message {
+	return New(content, origin, status, "", 0)
 }
 
-// NewMessage returns a new message
-func NewMessage(content []byte, origin *Origin, status string) *Message {
+// NewPartialMessage2 constructs message with content, status and log source.
+func NewPartialMessage2(content []byte, status string, source *config.LogSource) *Message {
+	return NewPartialMessage3(content, status, NewOrigin(source))
+}
+
+// NewPartialMessage constructs message with content, status and timestamp.
+func NewPartialMessage(content []byte, status string, timestamp string) *Message {
```

Comment:
Why do wee need to leak Partial Message here ?

Reply:
What do you mean by `leak Partial Message`? The original NewMessage is taking incomplete list of fields, which to me is already a partial message.